templates/daemonset.yaml

```diff
@@ -221,5 +227,9 @@ spec:
         - name: sd-shared
           emptyDir: {}
         {{- end }}
+        {{- if and .Values.child.claiming.enabled .Values.child.claiming.token .Values.child.claiming.rooms }}
+        - name: claim
+          emptyDir: {}
```

Comment:
https://github.com/netdata/helmchart/issues/79#issuecomment-656190514

do we need to persist data in `/var/lib/netdata/cloud.d`?

If so `emptyDir` doesnt do.

Reply:
That is my question

https://github.com/netdata/helmchart/issues/79#issuecomment-656187574

If we dont need to persist that data and having persistent `netdata.public.unique.id` is enough then `.Values.child.persistUniqueID` is an additional requirement for claiming.

Reply:
> remains the same through pod restarts

what about updates? delete/create? `emtpyDir` doesnt work, does it?

Reply:
Following the comment https://github.com/netdata/helmchart/pull/111#pullrequestreview-445726861 I tried to duplicate the `nodeuid` volume and volumeMount. 

Sorry for my ignorance, but if `/var/lib/netdata/cloud.d/claimed_id` stills the same after pod restarts why would there be necessity for persistence?

Reply:
> > remains the same through pod restarts
> 
> what about updates? delete/create? `emtpyDir` doesnt work, does it?

Sorry I used the wrong verb, I actually meant delete/create.
When the child pod running in a node is deleted and then recreated because of the DaemonSet, `/var/lib/netdata/cloud.d/claimed_id` remains the same. Should it be the same? should it be different?

Reply:
ok, did you check if it works with (for instance) image update (basically delete old pod and create a new one)? Is it still same _node_ in the cloud?

Reply:
> Sorry for my ignorance, but if /var/lib/netdata/cloud.d/claimed_id stills the same after pod restarts why would there be necessity for persistence?

To handle pod update. i am not sure having `netdata.public.unique.id` is enough.

Reply:
> ok, did you check if it works with (for instance) image update (basically delete old pod and create a new one)? Is it still same _node_ in the cloud?

Yes, I checked it, it stills the same _node_ in the cloud yet I encounter problem 1 mentioned in https://github.com/netdata/helmchart/pull/111#issuecomment-657094001

Reply:
according docs

>  When a Pod is removed from a node for any reason, the data in the emptyDir is deleted forever.

`/var/lib/netdata/cloud.d/claimed_id` should be deleted on the pod recreate.

if i understand it right `emptyDir` allows to survive only container restarts on crashes.

Reply:
ok, i think `.Values.child.persistUniqueID` should be another req for node claiming and should be added to the `if and ...`, because otherwise it would be a new node every time on a pod update.

Reply:
>  yet I encounter problem 1 mentioned in

i recall i have similar problems some time ago, and i had to restart netdata after install otherwise claiming was stuck in `Waiting for netdata to be claimed`.

Could you try to restart container? (like ssh to the host and docker kill `nd` child container)

Reply:
> according docs
> 
> > When a Pod is removed from a node for any reason, the data in the emptyDir is deleted forever.
> 
> `/var/lib/netdata/cloud.d/claimed_id` should be deleted on the pod recreate.
> 
> if i understand it right `emptyDir` allows to survive only container restarts on crashes.

I understand it the same way as you. And it makes me think then why this happens when the pod is deleted/created:
```
tdata-child-bsn8r netdata 2020-07-11 17:20:51: netdata INFO  : MAIN : Agent claiming command returned with code 13
netdata-child-bsn8r netdata 2020-07-11 17:20:51: netdata ERROR : MAIN : Agent failed to be claimed with the following error message:
netdata-child-bsn8r netdata 2020-07-11 17:20:51: netdata ERROR : MAIN : "already claimed" (errno 22, Invalid argument)
netdata-child-bsn8r netdata 2020-07-11 17:20:51: netdata INFO  : MAIN : File '/var/lib/netdata/cloud.d/claimed_id' was found. Setting state to AGENT_CLAIMED.
```

Maybe I should set the full file path `/var/lib/netdata/cloud.d/claimed_id` for the volumeMount using `subPath` ? :thinking:

Reply:
Well, i think it is expected, agent is already claimed indeed - it is same agent because it is same `netdata.public.unique.id`.

So claiming script returns an error (already claimed) but still it creates `claimed_id`.

Reply:
>  and nodes are manually removed from netdata cloud

if i understand it right it doesnt do much, because there is no unclaiming functionality.

Reply:
> > yet I encounter problem 1 mentioned in
> 
> i recall i had similar problems some time ago, and i had to restart netdata after install otherwise claiming was stuck in `Waiting for netdata to be claimed`.
> 
> Could you try to restart container? (like ssh to the host and docker kill `nd` child container)

Done. Not good results :cry: 

in node:
```
admin@ip-10-30-4-83:~$ sudo docker ps | grep netdata/netdata
2978e72849c8        netdata/netdata                                                  "/usr/sbin/run.sh"       15 minutes ago      Up 15 minutes                           k8s_netdata_netdata-parent-0_kube-system_974a931b-f338-4542-a50c-ba369f243857_0
cd50ab0db7fa        netdata/netdata                                                  "sh -c '/usr/sbin/ru…"   15 minutes ago      Up 15 minutes                           k8s_netdata_netdata-child-bsn8r_kube-system_6afb16e6-df90-43bb-b441-de9efe04d570_0
```
```
admin@ip-10-30-4-83:~$ sudo docker kill cd50ab0db7fa
cd50ab0db7fa
```

in the cluster:
```
netdata-child-bsn8r                                                 1/2     Error     0          18m
netdata-child-bsn8r                                                 1/2     Running   1          18m
netdata-child-bsn8r                                                 2/2     Running   1          18m
```

Yet the node remains unrecheable:
![image](https://user-images.githubusercontent.com/17387387/87230259-05124d00-c3af-11ea-846e-2b13cab7f37f.png)

Do you want me to attach a pastebin from the pod logs?